File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import (
    "strings"
)

func Score(word string) int {
    result := 0
    for _, letter := range word {
        result += CheckLetter(letter)
    }
    return result
}

func CheckLetter(letter rune) int {
    letter = rune(strings.ToUpper(string(letter))[0])

    switch {
    case strings.ContainsRune("AEIOULNRST", letter):
        return 1
    case strings.ContainsRune("DG", letter):
        return 2
    case strings.ContainsRune("BCMP", letter):
        return 3
    case strings.ContainsRune("FHVWY", letter):
        return 4
    case strings.ContainsRune("K", letter):
        return 5
    case strings.ContainsRune("JX", letter):
        return 8
    case strings.ContainsRune("QZ", letter):
        return 10
    default:
        return 0
    }
}

File: solutions/csharp/scrabble-score/1/ScrabbleScore.cs
public static class ScrabbleScore
{
    public static int Score(string input)
    {
        int result = 0;
        foreach (char ch in input)
        {
            result += CheckLetter(ch);
        }
        return result;
    }

    public static int CheckLetter(char letter)
    {
        letter = Char.ToUpper(letter);

        if ("AEIOULNRST".Contains(letter)) return 1;
        if ("DG".Contains(letter)) return 2;
        if ("BCMP".Contains(letter)) return 3;
        if ("FHVWY".Contains(letter)) return 4;
        if ("K".Contains(letter)) return 5;
        if ("JX".Contains(letter)) return 8;
        if ("QZ".Contains(letter)) return 10;
        return 0;
    }
}